### ISSUE TITLE
Using VarDumper::dumpAsString() instead var_export() in \yii\helpers\BaseVarDumper

### ISSUE THREAD

**Issue by XzAeRo** [yiisoft/yii2#8723]
When handling an undefined variable error, the debugger shows: `Warning: var_export does not handle circular references in ~/vendor/yiisoft/yii2/helpers/BaseVarDumper.php on line 187` at the bottom of the debugger page.

Looking for an answer I found an issue (#3327), where adresses a similar problem but for another file, and it was fixed properly by changing `var_export()` to `VarDumper::dumpAsString()`.

I tested the change and the error is gone.

Environment used:
* nginx/1.4.6
* Yii Framework/2.0.5-dev
* PHP 5.5.9-1ubuntu4.9 (php5-fpm)

**Comment by klimov-paul** [comment]
Confirmed. Unit test to reproduce:
```php
public function testExportRecursive()
    {
        $master = new \StdClass();
        $slave = new \StdClass();
        $master->slave = $slave;
        $slave->master = $master;
        $master->function = function() {return true;};

        $exportResult = VarDumper::export($master);
        $this->assertNotEmpty($exportResult);
    }
```

**Comment by samdark** [comment]
The fix is not correct since `dumpAsString` produces output that can't be evaluated by PHP.

**Comment by samdark** [comment]
The issue confirmed and test is OK.

**Comment by XzAeRo** [comment]
[This implementation may be useful](https://github.com/doctrine/common/blob/master/lib/Doctrine/Common/Util/Debug.php)

**Comment by klimov-paul** [comment]
Resolved by commit 16e1e8f

**Comment by klimov-paul** [comment]
d9eb660
